middleware/webhooks.js

```diff
@@ -7,9 +7,16 @@ module.exports = function createWithWebhook({ secret, shopStore }) {
     const topic = request.get('X-Shopify-Topic');
     const shopDomain = request.get('X-Shopify-Shop-Domain');
 
+    // Shopify escapes forward slashes 
+    // + replaces '&' with \u0026 when the HMAC is built
+    // so we need to do the same otherwise the validation will fail
+    let message = JSON.stringify(data)
+    message = message.split('/').join('\\/');
+    message = message.split('&').join('\\u0026');
```

Comment:
While Shopify does indeed replace `&` with `\u0026`, I don't see why this should be need to be done – the application should be computing the HMAC based on the body provided by Shopify, and Shopify has already performed this substitution.

Reply:
You're right @jamiemtdwyer but I have tested and computing the HMAC fails without these corrections...